src/App.tsx

```diff
@@ -15,36 +17,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+interface SortingParams {
+  sortField: string;
+  isReversed: boolean;
+}
+
+enum SortType {
+  Alphabet = 'ALPHABET',
+  Length = 'LENGTH',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: SortingParams,
+) {
+  const prepareGoods = [...goods];
+
+  if (sortField) {
+    prepareGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Alphabet:
+          return good1.localeCompare(good2);
+        case SortType.Length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.Alphabet,
+          })}
+          onClick={() => setSortField(SortType.Alphabet)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortType.Length,
+          })}
+          onClick={() => setSortField(SortType.Length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', { 'is-light': !isReversed })}
+          onClick={() => setIsReversed(!isReversed)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField('');
+              setIsReversed(false);
+            }}
```

Comment:
this can be moved to a separate function

Example:
```
const resetFilters = () => {
    setSortField(SortType.Default);
    setIsReversed(false);
};
```

```suggestion
            onClick={resetFilters}
```